packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.scss

```diff
@@ -7,189 +7,191 @@
 // --------------------------------------------------
 // MessageBar Styles
 
-:global {
-  .ms-MessageBar {
-    @include ms-baseFont;
-    @include ms-bgColor-info;
-    color: $ms-color-infoText;
-    width: 100%;
-    box-sizing: border-box;
-    display: flex;
-    position: relative;
-
-    .ms-MessageBar-icon {
-      @include padding-right(8px);
-      color: $ms-color-info;
-    }
-
-    .ms-Link {
-      font-size: $ms-font-size-s;
-    }
-  }
+.root {
+  @include ms-baseFont;
+  @include ms-bgColor-info;
+  color: $ms-color-infoText;
+  width: 100%;
+  box-sizing: border-box;
+  display: flex;
+  position: relative;
 
-  .ms-MessageBar-icon,
-  .ms-MessageBar-text {
-    display: table-cell;
-    vertical-align: top;
+  .icon {
+    @include padding-right(8px);
+    color: $ms-color-info;
   }
 
-  .ms-MessageBar-icon {
-    font-size: $ms-icon-size-m;
-    min-width: 16px;
-    min-height: 16px;
-    display: flex;
+  :global(.ms-Link) {
+    font-size: $ms-font-size-s;
   }
 
-  .ms-MessageBar-text {
-    @include ms-font-s;
-    min-width: 0;
-    display: flex;
+  :global(.ms-Button) {
+    @include margin-left(6px);
   }
+}
 
+.icon,
+.text {
+  display: table-cell;
+  vertical-align: top;
+}
 
-  //== Modifier: Warning message
-  //
-  .ms-MessageBar.ms-MessageBar--warning {
-    @include ms-bgColor-warning;
-    color: $ms-color-warningText;
-  }
+.icon {
+  font-size: $ms-icon-size-m;
+  min-width: 16px;
+  min-height: 16px;
+  display: flex;
+}
 
+.text {
+  @include ms-font-s;
+  min-width: 0;
+  display: flex;
+}
 
-  //== Modifier: Severe Warning message
-  //
-  .ms-MessageBar.ms-MessageBar--severeWarning {
-    @include ms-bgColor-severeWarning;
-    color: $ms-color-severeWarningText;
 
-    .ms-MessageBar-icon {
-      @include ms-fontColor-severeWarning;
-    }
-  }
+//== Modifier: Warning message
+//
+.root.rootIsWarning {
+  @include ms-bgColor-warning;
+  color: $ms-color-warningText;
+}
 
 
-  //== Modifier: Error message
-  //
-  .ms-MessageBar.ms-MessageBar--error {
-    @include ms-bgColor-error;
-    color: $ms-color-errorText;
+//== Modifier: Severe Warning message
+//
+.root.rootIsSevereWarning {
```

Comment:
Both classes are only ever applied to the root div of the control so in the current implementation they will show up together. I don't know if there was every any other intention but I don't want to change it since it's beyond the scope of the PR which is to modulify the sass.